pkg/reader/usdc_reader_test.go

```diff
@@ -0,0 +1,360 @@
+	emptyChainCCTP := CCTPDestDomains[uint64(emptyChain)]
+	emptyReader := reader.NewMockContractReaderFacade(t)
+	emptyReader.EXPECT().Bind(mock.Anything, mock.Anything).Return(nil)
+	emptyReader.EXPECT().QueryKey(
+		mock.Anything,
+		mock.Anything,
+		mock.Anything,
+		mock.Anything,
+		mock.Anything,
+	).Return([]types.Sequence{}, nil).Maybe()
+
+	faultyChain := cciptypes.ChainSelector(sel.AVALANCHE_MAINNET.Selector)
+	faultyReader := reader.NewMockContractReaderFacade(t)
+	faultyReader.EXPECT().Bind(mock.Anything, mock.Anything).Return(nil)
+	faultyReader.EXPECT().QueryKey(
+		mock.Anything,
+		mock.Anything,
+		mock.Anything,
+		mock.Anything,
+		mock.Anything,
+	).Return(nil, errors.New("error")).Maybe()
+
+	validSequence := []types.Sequence{
+		{
+			Data: &MessageSentEvent{Arg0: make([]byte, 128)},
+		},
+	}
+
+	validChain := cciptypes.ChainSelector(sel.ETHEREUM_MAINNET_ARBITRUM_1.Selector)
+	validReader := reader.NewMockContractReaderFacade(t)
+	validReader.EXPECT().Bind(mock.Anything, mock.Anything).Return(nil)
+	validReader.EXPECT().QueryKey(
+		mock.Anything,
+		mock.Anything,
+		mock.Anything,
+		mock.Anything,
+		mock.Anything,
+	).Return(validSequence, nil).Maybe()
+
+	tokensConfigs := map[cciptypes.ChainSelector]pluginconfig.USDCCCTPTokenConfig{
+		faultyChain: {
+			SourcePoolAddress:            "0x000000000000000000000000000000000000000A",
+			SourceMessageTransmitterAddr: "0x000000000000000000000000000000000000000B",
+		},
+		emptyChain: {
+			SourcePoolAddress:            "0x000000000000000000000000000000000000000C",
+			SourceMessageTransmitterAddr: "0x000000000000000000000000000000000000000D",
+		},
+		validChain: {
+			SourcePoolAddress:            "0x000000000000000000000000000000000000000E",
+			SourceMessageTransmitterAddr: "0x000000000000000000000000000000000000000F",
+		},
+	}
+
+	contactReaders := map[cciptypes.ChainSelector]contractreader.ContractReaderFacade{
+		faultyChain: faultyReader,
+		emptyChain:  emptyReader,
+		validChain:  validReader,
+	}
+
+	tokens := map[exectypes.MessageTokenID]cciptypes.RampTokenAmount{
+		exectypes.NewMessageTokenID(1, 1): {
+			ExtraData: NewSourceTokenDataPayload(11, emptyChainCCTP).ToBytes(),
+		},
+	}
+
+	usdcReader, err := NewUSDCMessageReader(tokensConfigs, contactReaders)
+	require.NoError(t, err)
+
+	tt := []struct {
+		name           string
+		sourceSelector cciptypes.ChainSelector
+		destSelector   cciptypes.ChainSelector
+		expectedMsgIDs []exectypes.MessageTokenID
+		errorMessage   string
+	}{
+		{
+			name:           "should return empty dataset when chain doesn't have events",
+			sourceSelector: emptyChain,
+			destSelector:   faultyChain,
+			expectedMsgIDs: []exectypes.MessageTokenID{},
+		},
+		{
+			name:           "should return error when chain reader errors",
+			sourceSelector: faultyChain,
+			destSelector:   emptyChain,
+			errorMessage:   "error querying contract reader for chain 6433500567565415381",
```

Comment:
Nit: use format and use the chain selector variable from faultyChain instead of hardcoding.